sequencer/src/block2.rs

```diff
@@ -190,13 +218,140 @@ pub struct TxInclusionProof {
 
 #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
 enum TxTableRangeProof {
-    // If this is the first tx then we need only 1 proof `end` index.
+    // If this is the first tx then we need only 1 proof for `end` index.
     First(RangeProof),
     // All other cases need 2 proofs: 1 for `start`, 1 for `end`.
     Other(RangeProof, RangeProof),
     // TODO new variant `Final` with implicit `end` https://github.com/EspressoSystems/espresso-sequencer/issues/757
 }
 
+impl TxInclusionProof {
+    // TODO(795) prototype only!
+    //
+    // - Obnoxious arg list; where to store all this cruft?
+    // - Returns `None` if an error occurred.
+    // - Use of `Result<(),()>` pattern to indicate a must-use bool return type.
```

Comment:
I don't mind `Result<(), ()>` but FYI you can also put `#[must_use]` on a function returning `bool` (or anything)

Reply:
True. I forgot to mention that it also enables use of `?` for concise abort-on-failure.

Reply:
Oh yeah that's nice